quickbudget/views.py

```diff
@@ -0,0 +1,162 @@
+from rest_framework import generics, status, filters
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from quickbudget.api.permissions import (
+    BudgetMembersOnly,
+    AllBudgetExpenseMembersOnly,
+    ExpenseBudgetMembersOnly, IsPostOrIsAuthenticated,
+)
+from quickbudget.api.serializers import (
+    ExpenseSerializer,
+    CategoriesSerializer,
+    BudgetSerializer,
+    MemberAddSerializer,
+    RemoveBudgetMembersSerializer,
+    UserSerializer,
+)
+from quickbudget.models import Budget, Expense, Category, User
+
+
+class ListAddBudgets(generics.ListCreateAPIView):
+    queryset = Budget.objects.all()
+    serializer_class = BudgetSerializer
+
+    def perform_create(self, serializer):
+        # Set the creator field to the current authenticated user
+        serializer.validated_data['creator'] = self.request.user
+
+        if serializer.is_valid():
+            # Create the budget object
+            budget = serializer.save()
+
+            # Add the current user as a member of the budget
+            budget.members.add(self.request.user)
+
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+        return Response(serializer.data, status=status.HTTP_400_BAD_REQUEST)
+
+    def get_queryset(self):
+        return Budget.objects.filter(members=self.request.user)
+
+
+class BudgetDetail(generics.RetrieveUpdateDestroyAPIView):
+    queryset = Budget.objects.all()
```

Comment:
copying it from your previous PR:
```
you need to be careful when usiing `queryset = Budget.objects.all()`. in some cases, you'd need to overwrite `get_queryset` method and filter data.
```

try to debug what data is in there, and what data you expect.

Reply:
It was displaying all the data and with get_queryset we can filter for data that can be viewed by currently logged in user.